### ISSUE TITLE
Problem with 500 responses and redirect: 'manual'

### ISSUE THREAD

**Issue by pekeler** [bitinn/node-fetch#103]
This works:
```JavaScript
fetch('http://www.mocky.io/v2/570d38222700005801dc29ba', {redirect: 'error'}).then(function(response) {console.log('good');}).catch(function(error) {console.error('bad');});
```

This breaks:
```JavaScript
fetch('http://www.mocky.io/v2/570d38222700005801dc29ba', {redirect: 'manual'}).then(function(response) {console.log('good');}).catch(function(error) {console.error('bad');});

> TypeError: Parameter 'url' must be a string, not object
    at Url.parse (url.js:90:11)
    at urlParse (url.js:84:5)
    at Url.resolve (url.js:426:29)
    at urlResolve (url.js:422:40)
    at ClientRequest.<anonymous> (/Users/pekeler/Projects/wdt-remote-kicker/dev/remote_kicker/node_modules/node-fetch/index.js:179:29)
    at emitOne (events.js:77:13)
    at ClientRequest.emit (events.js:169:7)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:414:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:88:23)
    at Socket.socketOnData (_http_client.js:304:20)
```

cc @jimmywarting

**Comment by pekeler** [comment]
Just change https://github.com/bitinn/node-fetch/blob/1cd0a5c57047213a902efcdb26055e0af603d5da/index.js#L178 to

```JavaScript
if (self.isRedirect(res.statusCode) &&  options.redirect === 'manual') {
```
?

**Comment by bitinn** [comment]
I don't think that's the cause for your problem, I failed to see why you are passing an object as url.

We do have test cases to verify this works.

https://github.com/bitinn/node-fetch/blob/master/test/test.js#L301

**Comment by bitinn** [comment]
Ah got it, would be better to check if `location` header exists.

**Comment by bitinn** [comment]
fixed and released as v1.5.1, must be a new record for us.

**Comment by pekeler** [comment]
Awesome - Thanks!
